app/controllers/katello/concerns/api/v2/registration_controller_extensions.rb

```diff
@@ -35,14 +35,28 @@ def context_urls
 
       def smart_proxy
         @smart_proxy ||= begin
-          proxy = params[:url] ? SmartProxy.unscoped.find_by(url: params[:url]) : SmartProxy.pulp_primary
+          proxy = params[:url] ? find_smart_proxy_by_url : SmartProxy.pulp_primary
 
           fail Foreman::Exception, _('Smart proxy content source not found!') unless proxy
           fail Foreman::Exception, _('Pulp 3 is not enabled on Smart proxy!') unless proxy.pulp3_enabled?
 
           proxy
         end
       end
+
+      # Find proxy by URL configured in templates module
+      # or by proxy's URL
+      def find_smart_proxy_by_url
+        feature = Feature.find_by(name: 'Templates')
+
+        if feature
```

Comment:
I think this logic doesn't work since the feature will always exist, even if no Smart Proxies using it exist. So it will never fall back to the `else` branch.